server/load.go

```diff
@@ -0,0 +1,81 @@
+package server
+
+import (
+	"math"
+	"os"
+	"sync/atomic"
+	"time"
+
+	sigar "github.com/cloudfoundry/gosigar"
+)
+
+// Load represents the system resources that the server process is consuming,
+// in terms of CPU% and megabytes of memory.
+type Load struct {
+	currentCPULoad    uint32
+	currentMemoryLoad uint32
+	done              chan bool
+	pid               int
+	pm                *sigar.ProcMem
+	pc                *sigar.ProcCpu
+}
+
+// StartLoadMonitoring begins the load monitoring.  The monitor can be stopped
+// by invoking load.Close()
+func StartLoadMonitoring() *Load {
+	l := &Load{
+		done: make(chan bool),
+		pid:  os.Getpid(),
+		pm:   &sigar.ProcMem{},
+		pc:   &sigar.ProcCpu{},
+	}
+
+	go func() {
+		for {
+			select {
+			case <-l.done:
+				break
+			case <-time.After(1 * time.Second):
+				l.update()
+			}
+		}
+	}()
+
+	return l
+}
+
+// Close will end the load monitoring.
+func (l *Load) Close() error {
+	l.done <- true
+	return nil
+}
+
+// CPU reports the instantaneous CPU load.
+func (l *Load) CPU() float32 {
```

Comment:
Might consider a single `Stats` API that returns a `float32` and an `uint32` for CPU and memory load, as these are unlikely to not be called together.